docs/api_compatibility.md

```diff
@@ -0,0 +1,47 @@
+# API compatibility
+
+JAX is constantly evolving, and we are not yet ready to commit to a stable API.
+That said, we want to minimize churn for the JAX user community.
+
+JAX follows a 3 month deprecation policy. When an incompatible change is made
+to an API, we will make our best effort to obey the following procedure:
+* the change will be announced in `CHANGELOG.md`, and the old API will issue a
+  `DeprecationWarning`.
```

Comment:
It's worth noting that DeprecationWarning is often silenced by default!

FutureWarning is a better choice for something you want end-users to see.

Reply:
PEP565 indicates that one should issue `DeprecationWarning` if the target of the warning is other python developers. Am I misunderstanding it or is it incorrect in practice?